bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -0,0 +1,147 @@
+        y = kde.predict(x_grid)
+    else:
+        y = kde.predict(x_grid[:, np.newaxis])
+    if ret_kde == True:
+        return kde, y
+    return y
+
+def kfcv_awkde(sample, bwchoice, alphachoice, k=2):
+    if bwchoice not in ['silverman', 'scott']:
+        bwchoice = float(bwchoice)
+    fomlist =[]
+
+    #split data into two subset of equal size [or almost equal (n+1, n) parts]
+    kfold = KFold(k, True, None)
+    for train_index, test_index in kfold.split(sample):
+        train, test = sample[train_index], sample[test_index]
+        y = kde_awkde(train, test, alp=alphachoice, gl_bandwidth=bwchoice)
+        fomlist.append(np.sum(np.log(y)))
+    return np.mean(fomlist)
+
+def kfcv_awkdeOnedim(sample, bwchoice, alphachoice, k=2):
+    if bwchoice not in ['silverman', 'scott']:
+        bwchoice = float(bwchoice)
+    fomlist =[]
+    kfold = KFold(k,True,None)
+    for train, test in kfold.split(sample):
+        y = kdeOnedim(train, test, alp=alphachoice, gl_bandwidth=bwchoice)
+        fomlist.append(np.log(y))
+    return np.mean(fomlist)
+
+def optimizedparam(sampleval):
+    bwgrid = ['scott', 'silverman']+np.logspace(-2,0,10).tolist()
+    alphagrid = [1]#np.linspace(0.1, 1, 10)
+    FOM= {}
+    FOMxval = {}
+    FOMyval = {}
+    for gbw in bwgrid:
+        for alphavals in alphagrid:
+            FOM[(gbw, alphavals)] = kfcv_awkde(sampleval, gbw, alphavals)
+    optval = max(FOM.items(), key=operator.itemgetter(1))[0]
+    optbw, optalpha  = optval[0], optval[1]
+    maxFOM = FOM[(optbw, optalpha)]
+    return optbw, optalpha
+
+#calling template data and constructing template_kde_hdf_file_over_same_template_params
+template_file = h5py.File(args.template_file, 'r')
+mass1_n = template_file['mass1'][:]
+mass2_n = template_file['mass2'][:]
+spin1z_n = template_file['spin1z'][:]
+spin2z_n = template_file['spin2z'][:]
+chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
+mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
+eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+
+if args.make_template_kde:
+    sample_n = np.vstack((np.log(mchirp_n), eta_n, chi_eff_n)).T
+    grid_pts_n = np.array((np.log(mchirp_n), eta_n, chi_eff_n)).T
+    optbw_n, optalpha_n = optimizedparam(sample_n)
+    kde_data_n = kde_awkde(sample_n, grid_pts_n, alp=optalpha_n, gl_bandwidth=optbw_n)
+
+    hdf = h5py.File('kdevals_template.hdf', 'w')
+    hdf.create_dataset('kde_template', data=kde_data_n)
+    hdf.attrs['bank_kde'] = 'kde_template'
+    hdf.create_dataset('mass1', data=mass1_n)
+    hdf.create_dataset('mass2', data=mass2_n)
+    hdf.create_dataset('spin1z', data=spin1z_n)
+    hdf.create_dataset('spin2z', data=spin2z_n)
+    hdf.create_dataset('mchirp', data=mchirp_n)
+    hdf.create_dataset('eta', data=eta_n)
+    hdf.create_dataset('chi_eff', data=chi_eff_n)
+    hdf.create_dataset('grid_pts', data=grid_pts_n)
+    hdf.close()
+
+#calling signal data and constructing signal_kde_hdf_file_over_template_params
```

Comment:
I think you only want to do this if the user has specified to make the signal KDE, otherwise you don't need to.  So put these lines inside an 'if' statement